commands/operator-sdk/cmd/test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	testLocation     string
+	kubeconfig       string
+	crdManifestPath  string
+	opManifestPath   string
+	rbacManifestPath string
+	verbose          bool
+)
+
+func NewTestCmd() *cobra.Command {
+	testCmd := &cobra.Command{
+		Use:   "test --test-location <path to tests directory> [flags]",
+		Short: "Run End-To-End tests",
+		Run:   testFunc,
+	}
+	defaultKubeConfig := ""
+	homedir, ok := os.LookupEnv("HOME")
+	if ok {
+		defaultKubeConfig = homedir + "/.kube/config"
+	}
+	testCmd.Flags().StringVarP(&testLocation, "test-location", "t", "", "Location of test files (e.g. ./test/e2e/)")
+	testCmd.MarkFlagRequired("test-location")
+	testCmd.Flags().StringVarP(&kubeconfig, "kubeconfig", "k", defaultKubeConfig, "Kubeconfig path")
+	testCmd.Flags().StringVarP(&crdManifestPath, "crd", "c", "deploy/crd.yaml", "Path to CRD manifest")
+	testCmd.Flags().StringVarP(&opManifestPath, "operator", "o", "deploy/operator.yaml", "Path to operator manifest")
+	testCmd.Flags().StringVarP(&rbacManifestPath, "rbac", "r", "deploy/rbac.yaml", "Path to RBAC manifest")
+	testCmd.Flags().BoolVarP(&verbose, "verbose", "v", false, "Enable verbose go test")
+
+	return testCmd
+}
+
+func testFunc(cmd *cobra.Command, args []string) {
+	testArgs := []string{"test", testLocation + "/..."}
+	if verbose {
+		testArgs = append(testArgs, "-v")
```

Comment:
Consider allowing arbitrary passthrough of test flags, perhaps following the typical convention (e.g. `go test` itself) of "everything after [--|-args] is passed through".

Reply:
I feel like that would be quite difficult to do; I'm currently thinking of ways to implement that in a simple manner. The way `go test` does it is quite complicated and they don't use the flag package for `go test`, instead opting for separate code: https://github.com/golang/go/blob/master/src/cmd/go/internal/test/testflag.go#L21

Reply:
I'll add that as a TODO for now. It would be nice to be able to do it with cobra, but if that's too complicated, we may just implement it as a string that the user passes in (e.g. `operator-sdk test -t ./test/e2e/ --passthrough "-v -parallel=2"`.

Reply:
Sure... it may be useful to help me understand the design goals. For example, if a goal is to provide a simple wrapper which works for most cases but consistently provides an escape hatch for advanced uses (via the standard Go toolchain) then just keeping what you have here seems reasonable to me. If the wrapper is intended to be the *only* reasonable way to use the test framework, then things get a little trickier in deciding how much of the standard toolchain to re-expose, and how.

Reply:
@ironcladlou It's the former. The wrapper is meant to be a simple way for users to run their e2e tests in an SDK based project. I don't think there's any restriction for it to be the only way.
For more advanced use cases `go test` could be used.